### DISCUSSION TITLE
Shields no longer visible in v0.10 endless sky?

### DISCUSSION THREAD

**Discussion by ghost** [endless-sky/endless-sky#8614] [Q&A] [answered]
Hello,

This is my first time posting on this forum for this game, so I am uncertain I have this post in the proper place.  My apologies in advance if I do not.

In the latest major endless sky release, v10, are the shields on the ships (especially pirate ships, but my own as well), no longer visible?  Am I missing a setting or something that could turn them on and leave them on during the game?

I ask because I have some vision issues and the visibility of the shields in the space field all but make it essential for me to find enemy ships in the star fields.  Without them, the game is more frustration then fun, which means I may have either revert to an older version (and miss out on many things that I consider improvements in the game insofar as I read in the changelog), or abandon it altogether.  

Also, in regard to the lower left corner information about commodity prices, is there a way to slightly enlarge the lower left corner letters and numbers, and possibly have a brighter contrast of letters and background?  This, too, is a vision issue due to the letter size and the letters and background being too close in tones.  Brighter white letters, possibly highlighted brighter white still, and a little larger would be a help.  If there is a plug-in that could allow this adjustment, I would be happy to give it a try.  

Again, I do not know if this is the right place to post these two concerns.  If not, please let me know where it should have been posted and feel free to move it to that location and let me know of the move.  

Thanks in advance

Star Penguin

**Comment by quyykk** [comment, marked as answer]
> are the shields on the ships (especially pirate ships, but my own as well), no longer visible?

Do you mean these?

![image](https://user-images.githubusercontent.com/85879619/229186171-8bfc779b-7260-4d5a-9e7a-a6b4c18bd69c.png)

> Also, in regard to the lower left corner information about commodity prices, is there a way to slightly enlarge the lower left corner letters and numbers, and possibly have a brighter contrast of letters and background? This, too, is a vision issue due to the letter size and the letters and background being too close in tones. Brighter white letters, possibly highlighted brighter white still, and a little larger would be a help. If there is a plug-in that could allow this adjustment, I would be happy to give it a try.

Unfortunately it is not possible to increase the font size, however you can increase the general zoom in the preferences ("Main zoom factor").

> Again, I do not know if this is the right place to post these two concerns. If not, please let me know where it should have been posted and feel free to move it to that location and let me know of the move.

Yes here is fine, thanks! :)

**Reply by ghost** [reply]
To the ship picture, yes, those shields.  The visibility of the green circular shields on my ships and the complementary orange shields on the enemy/pirate ships have been a great help for me seeing what is going on in the space fields  I did not see shields on the ships in the star field, though, until I just now zoomed in and found hues of green on my ship... at a close zoom.  Sorry to say, while it is a nice and appropriate effect (objectively, it makes sense, because that is how you would see it in television shows and movies), it is not helpful for me.  Did I miss a setting or something here or is this one of the recent changes, and did I miss that notice somewhere?

As to the main zoom setting, I do not know how, but I did miss seeing it in the preferences.  Sigh.  Not going to say how long I have been playing without noticing that.  Anyway, I have found it now and, yes, I can work with that.  So, one concern settled.  Thank you for that help.  :+1:

**Reply by quyykk** [reply]
Those shields should still be there in the settings, it's called "status overlays". I'm not aware of any change that would have disabled it or something? So maybe check again if it's enabled.

**Reply by warp-core** [reply]
![image](https://user-images.githubusercontent.com/20605679/229193847-0ecd0ab9-71ed-4cd3-98ed-5f60521b28a6.png)

**Reply by ghost** [reply]
Thank you for the reply.  The setting was off.  It is on now.  And it is working now.  Thanks for your help on these questions.  

Keep up the great work on this evolving project / game.

**Comment by warp-core** [comment]
Regarding text colours, most of that should be customisable.
Most text should use colours defined in the `data/interfaces.txt` file.
Specifically:
```
# Colors generally used for drawing text or lines.
color "bright" .75 .75 .75 0.
color "medium" .5 .5 .5 0.
color "dim" .25 .25 .25 0.
color "dimmer" .18 .18 .18 0.
color "faint" .1 .1 .1 0.
color "dark" .05 .05 .05 0.
```

Each value represents the red, green, blue, and alpha (opacity) component of hte given named colour, on a scale from 0 to 1.

So, for example, changing the `bright` colour to `1. 1. 1. 0.` gives:

.75 | 1.
-- | --
![image](https://user-images.githubusercontent.com/20605679/229195783-a5372a4b-ad49-447a-b800-3dcdcad7c270.png) | ![image](https://user-images.githubusercontent.com/20605679/229196094-3bab6035-9be4-4e14-afd2-06ca660dca21.png)
![image](https://user-images.githubusercontent.com/20605679/229196399-7d78cc20-d98a-44b8-866a-e60970530f7c.png) | ![image](https://user-images.githubusercontent.com/20605679/229196266-ac283b99-f86a-47de-bc8a-b253a62cb3d7.png)

The background colour of the conversation panel can also be changed:

default | glorious darkness | maximum contrast
-- | -- | --
![image](https://user-images.githubusercontent.com/20605679/229197076-44e77c94-dcf4-4607-beff-4510c1cecb07.png) | ![image](https://user-images.githubusercontent.com/20605679/229197476-b93baac0-6b65-4efb-a06e-a4ebea5d9957.png) | ![image](https://user-images.githubusercontent.com/20605679/229197751-1ccdc71f-3c0f-4620-8c99-e3c8abd1a282.png)

The conversation background colour being:
`color "conversation background" .125 .125 .125 1.`
Also, normal conversation text uses the `medium` colour.

The third conversation screenshot was produced with the values:
```
color "bright" 1. 1. 1. 0.
color "medium" 1. 1. 1. 0.
color "conversation background" 0. 0. 0. 1.
```

Though, this eliminates the contrast between the conversation text and the text of choices, and likely removes contrast between different text in many other places to, so it may be best not to use the same values for everything.

It is not currently possible to change the background colour of most other panels except by modifying the images they use, which is not as simple as setting some new configuration values.

If you want to modify these colours yourself, you can do so by editing the file in the game data directly, though this is generally not a recommended way of editing data, as your changes will be overwritten when you update.
The better way to do so is to create a plugin, see [the creating plugins wiki page](https://github.com/endless-sky/endless-sky/wiki/CreatingPlugins) for the basic structure.
Once you have that, the colour definition will need to go in a text file in the data subfolder of the plugin (the name of the file doesn't matter as long as it is a plain text (`.txt`) file.)

Many other colours, including the colours used by the status overlays, are also customisable in this way.

If you need any help ffiguring out where in game some colour is used or what the name of the colours used for something in game is, feel free to ask.
